neuralmonkey/tf_manager.py

```diff
@@ -173,14 +174,21 @@ def execute(self,
                 train=False,
                 compute_losses=True,
                 summaries=True,
-                batch_size=None) -> List[ExecutionResult]:
+                batch_size=None,
+                log_progress: int = 0) -> List[ExecutionResult]:
         if batch_size is None:
             batch_size = len(dataset)
         batched_dataset = dataset.batch_dataset(batch_size)
+        log_time = time.process_time()
 
         batch_results = [
             [] for _ in execution_scripts]  # type: List[List[ExecutionResult]]
-        for batch in batched_dataset:
+        for batch_id, batch in enumerate(batched_dataset):
+            if (log_time + log_progress * 60 < time.process_time()
+                    and log_progress > 0):
```

Comment:
tohle mi přijde divně napsaný.. Když to řeknu přirozenym jazykem, tak ta podmínka je "pokud uplynulo 60 vteřin od minulýho logování", tj, pokud rozdíl času mezi teď a minulym je 60 vteřin, tj. spíš bych psal `if time.process_time() - log_time > log_progress` a přejmenoval bych `log_time` na `prev-` nebo `last_log_time`

A taky bych dal podmínku `log_progress > 0` napřed, protože je jednodušší (pokud to náhodou ta první neni ta dlouhá, kterou ti nešlo rozseknout :-) )